test/integration/document_collection_test.rb

```diff
@@ -34,18 +24,47 @@ class DocumentCollectionTest < ActionDispatch::IntegrationTest
     assert page.has_text?("Each regime page provides a current list of asset freeze targets designated by the United Nations (UN), European Union and United Kingdom, under legislation relating to current financial sanctions regimes.")
   end
 
-  test "renders contents with link to each collection group" do
+  test "renders a contents list for each content item for every document collection group" do
     setup_and_visit_content_item("document_collection")
+    assert_equal 6, @content_item["details"]["collection_groups"].size
+
     @content_item["details"]["collection_groups"].each do |group|
       assert page.has_css?("nav a", text: group["title"])
     end
   end
 
-  test "renders a content list" do
+  test "renders a content list with 6 items, ignoring any empty groups" do
     setup_and_visit_content_item("document_collection")
+    @content_item["details"]["collection_groups"] << { "title" => "Empty Group", "documents" => [] }
+
+    visible_items = all("nav.gem-c-contents-list .gem-c-contents-list__list-item")
+    assert_equal 6, visible_items.size
+
+    @content_item["details"]["collection_groups"].each do |group|
+      next if group["documents"].empty?
+
+      assert page.has_css?("nav a", text: group["title"])
+    end
+
     assert page.has_css?(".gem-c-contents-list", text: "Contents")
   end
 
+  test "renders no contents list if document collection groups are empty, even if a group has more than 415 characters" do
+    content_item = get_content_example("document_collection")
+
+    content_item["details"]["collection_groups"] = []
+
+    content_item["details"]["collection_groups"] << {
+      "title" => "Empty Group",
+      "documents" => [],
+      "body" => "<div><p>#{Faker::Lorem.characters(number: 416)}</p></div>",
+    }
+
+    stub_content_store_has_item(content_item["base_path"], content_item.to_json)
+    visit(content_item["base_path"])
+    assert_not page.has_css?(".gem-c-contents-list")
+  end
+
```

Comment:
I'm not sure if there is a better way to write this test case, but I had to add `return nil if groups.empty?` to the first_item method in the [document collection presenter](app/presenters/document_collection_presenter.rb) as I kept getting error messages when trying to create an empty group with no documents, but the body is populated.

The error message I received was: 

```
ActionView::Template::Error: undefined method `[]' for nil:NilClass
            app/presenters/document_collection_presenter.rb:98:in `first_item'
            app/presenters/content_item/contents_list.rb:103:in `no_first_item?'
            app/presenters/content_item/contents_list.rb:23:in `show_contents_list?'
            app/presenters/content_item/contents_list.rb:10:in `contents'
            app/views/content_items/document_collection.html.erb:32
            app/controllers/content_items_controller.rb:198:in `block in render_template'
            app/controllers/content_items_controller.rb:224:in `with_locale'
            app/controllers/content_items_controller.rb:197:in `render_template'
            app/controllers/content_items_controller.rb:36:in `show'
```

Reply:
@hannako